gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/service/InboundMessageMergingServiceTests.java

```diff
@@ -0,0 +1,575 @@
+
+    @InjectMocks
+    private InboundMessageMergingService inboundMessageMergingService;
+
+    @SneakyThrows
+    private void prepareMocks(InboundMessage inboundMessage, ArrayList<PatientAttachmentLog> attachments) {
+        var inboundMessageAsString = objectMapper.writeValueAsString(inboundMessage);
+        var patientMigrationRequest = PatientMigrationRequest
+                .builder()
+                .inboundMessage(inboundMessageAsString)
+                .build();
+
+        when(patientAttachmentLogService.findAttachmentLogs(CONVERSATION_ID)).thenReturn(attachments);
+        when(migrationRequestDao.getMigrationRequest(any())).thenReturn(patientMigrationRequest);
+        when(objectMapper.readValue(inboundMessageAsString, InboundMessage.class)).thenReturn(inboundMessage);
+        when(attachmentReferenceUpdaterService.updateReferenceToAttachment(any(), any(), any())).thenReturn(inboundMessage.getPayload());
+
+        if (attachments.get(0).getSkeleton()) {
+            prepareSkeletonMocks(inboundMessage);
+        }
+    }
+
+    private void prepareSkeletonMocks(InboundMessage inboundMessage) throws SAXException, TransformerException {
+
+        var reference = new EbxmlReference("First instance is always a payload", "mid:1", "docId");
+        var ebXmlAttachments = Arrays.asList(reference);
+        var fileAsBytes = readInboundMessagePayloadFromFile().getBytes(StandardCharsets.UTF_8);
+
+        when(attachmentHandlerService.getAttachment(FILENAME)).thenReturn(fileAsBytes);
+        when(xmlParseUtilService.getEbxmlAttachmentsData(inboundMessage)).thenReturn(ebXmlAttachments);
+        when(xPathService.parseDocumentFromXml(any())).thenReturn(ebXmlDocument);
+        when(xPathService.getNodes(any(), any())).thenReturn(nodeList);
+        when(nodeList.item(0)).thenReturn(node);
+        when(nodeList.item(1)).thenReturn(node);
+        when(ebXmlDocument.getElementsByTagName("*")).thenReturn(nodeList);
+        when(xmlParseUtilService.getStringFromDocument(any())).thenReturn(inboundMessage.getPayload());
+        when(node.getOwnerDocument()).thenReturn(ebXmlDocument);
+        when(node.getParentNode()).thenReturn(node);
+
+    }
+
+    @Test
+    public void When_HappyPathWithSkeleton_Expect_ThrowNoErrors() throws JAXBException, JsonProcessingException, TransformerException {
+        var inboundMessage = new InboundMessage();
+        var attachments = createPatientAttachmentList(true, true);
+
+        inboundMessage.setPayload("payload");
+        inboundMessage.setPayload(readInboundMessagePayloadFromFile());
+        inboundMessage.setEbXML(readInboundMessageEbXmlFromFile());
+
+        prepareMocks(inboundMessage, attachments);
+
+        inboundMessageMergingService.mergeAndBundleMessage(CONVERSATION_ID);
+
+        verify(nackAckPreparationService, never()).sendNackMessage((NACKReason) any(), (RCMRIN030000UK06Message) any(), any());
+        verify(xmlParseUtilService, times(1)).getStringFromDocument(any());
```

Comment:
It would be nice if either `when` or `verify` method checked real arguments, not `any()`. (This comment applies to other mocks as well)